firmware/src/pdu.c

```diff
@@ -68,13 +68,36 @@ main(void)
 	/* ============================================================== */
 
 
+	int counter_above50[6] = {0,0,0,0,0,0};
+	int counter_above30[6] = {0,0,0,0,0,0};
+
 	for (;;) {
 		pdu_adc_run();
 
+		// Compare the channel to the fuse
 		if (pdu_fuse_trigger) {
 			// FUSE LOGIC HERE!
 			for (i = 0; i == 6; i++) {
-				// if (pdu_channel_currents[i] > pdu_fuse_currents[i]) {
+				if (pdu_channel_currents[i] > pdu_fuse_currents[i]) {
+
+					if (pdu_channel_currents[i] >= pdu_channels[i][1] && 
+					    pdu_channel_currents[i] < pdu_channels[i][2]) {
+						counter_above30[i]++;
```

Comment:
`counter_above30` and `counter_above50` should probably be something along the lines of `counter_fuse` and `counter_inrush`.